sdk/ai/ai-projects/src/customization/convertOutputModelsFromWire.ts

```diff
@@ -0,0 +1,671 @@
+): MessageTextContentOutput {
+  return {
+    type: input.type,
+    text: convertMessageTextDetailsOutput(input.text),
+  };
+}
+
+export function convertMessageTextDetailsOutput(
+  input: WireMessageTextDetailsOutput
+): MessageTextDetailsOutput {
+  return {
+    value: input.value,
+    annotations: input.annotations?.map(convertMessageTextAnnotationOutput),
+  };
+}
+
+export function convertMessageTextAnnotationOutput(
+  input: WireMessageTextAnnotationOutputParent
+): MessageTextAnnotationOutput {
+  switch (input.type) {
+    case "file_citation":
+      return convertMessageTextFileCitationAnnotationOutput(
+        input as WireMessageTextFileCitationAnnotationOutput
+      );
+    case "file_path":
+      return convertMessageTextFilePathAnnotationOutput(
+        input as WireMessageTextFilePathAnnotationOutput
+      );
+    default:
+      return { ...input };
+  }
+}
+
+export function convertMessageTextFileCitationAnnotationOutput(
+  input: WireMessageTextFileCitationAnnotationOutput
+): MessageTextFileCitationAnnotationOutput {
+  return {
+    type: input.type,
+    text: input.text,
+    fileCitation: convertMessageTextFileCitationDetailsOutput(input.file_citation)
+  };
+}
+
+export function convertMessageTextFileCitationDetailsOutput(
+  input: WireMessageTextFileCitationDetailsOutput
+): MessageTextFileCitationDetailsOutput {
+  return {
+    fileId: input.file_id,
+    quote: input.quote
+  };
+}
+
+export function convertMessageTextFilePathAnnotationOutput(
+  input: WireMessageTextFilePathAnnotationOutput
+): MessageTextFilePathAnnotationOutput {
+  return {
+    type: input.type,
+    filePath: convertMessageTextFilePathDetailsOutput(input.file_path),
+    startIndex: input.start_index,
+    endIndex: input.end_index,
+    text: input.text,
+  };
+}
+
+export function convertMessageTextFilePathDetailsOutput(
+  input: WireMessageTextFilePathDetailsOutput
+): MessageTextFilePathDetailsOutput {
+  return { fileId: input.file_id };
+}
+
+export function convertMessageImageFileContentOutput(
+  input: WireMessageImageFileContentOutput
+): MessageImageFileContentOutput {
+  return {
+    type: input.type,
+    imageFile: convertMessageImageFileDetailsOutput(input.image_file)
+  };
+}
+
+export function convertMessageImageFileDetailsOutput(
+  input: WireMessageImageFileDetailsOutput
+): MessageImageFileDetailsOutput {
+  return { fileId: input.file_id };
+}
+
+export function convertThreadRunOutput(
+  input: WireThreadRunOutput
+): ThreadRunOutput {
+  return {
+    id: input.id,
+    object: input.object,
+    threadId: input.thread_id,
+    assistantId: input.assistant_id,
+    status: input.status,
+    requiredAction: input.required_action ? convertRequiredActionOutput(input.required_action) : undefined,
+    lastError: input.last_error,
+    model: input.model,
+    instructions: input.instructions,
+    tools: input.tools.map(convertToolDefinitionOutput),
+    createdAt: new Date(input.created_at * 1000),
```

Comment:
interesting that `createdAt` has a different precision than the other dates... definitely something that will be good to let the emitter handle eventually

Reply:
good catch, its error, I have fixed